### ISSUE TITLE
Utils.random(from, to) not within range

### ISSUE THREAD

**Issue by sassomedia** [amcharts/amcharts4#2779]
For `Utils.random`, there seems to be an incorrect or missed calculation in:
https://github.com/amcharts/amcharts4/blob/master/src/.internal/core/utils/Utils.ts#L833
This will most of the time go out of `to` range because there's no subtraction of the `from` value.
```javascript
return Math.floor(Math.random() * to) + from;
```

The right calculation would be:
```javascript
return Math.floor(Math.random() * (to - from)) + from;
```

**Comment by martynasma** [comment]
Thanks! This will be fixed in the next version. Will notify you here when it happens.

**Comment by sassomedia** [comment]
Sounds great, thanks!

**Comment by martynasma** [comment]
Fixed in [4.9.35](https://github.com/amcharts/amcharts4/releases/tag/4.9.35).

## [4.9.35] - 2020-08-12

### Added
- New adapter on `XYSeries`: `groupValue`. It's now possible to apply custom aggregate value calculation functions for grouped data items. [More info](https://www.amcharts.com/docs/v4/tutorials/using-custom-functions-for-data-item-grouping/).
- New adapter on `Cursor`: `cursorPoint`. Use it to custom-position chart cursor using your own logic.

### Changed
- Accessibility: Now if ENTER is pressed when object with an enabled `"hit"` event is focused, that event will be triggered automatically.
- Clicking anywhere on the plot area with `XYCursor` will now trigger blur on any focused element on the page (consistent to clicking anywhere else on the document).

### Fixed
- RangeSelector plugin was not properly inheriting chart's locale.
- `"x"` (timestamp) in `inputDateFormat` was not working properly.
- If all values of `SankeyDiagram` were `0` (zero), nodes were not being arranged and their rectangles were not being displayed.
- First `CategoryAxis` label was not visible if `minGridDistance` was set to `0` (zero).
- Old series bullets of an `XYSeries` remained visible when data was updated and data grouping was enabled on related `DateAxis`.
- `Utils.random()` function was not returning value in proper from/to range.
- Stacking now works only between series of the same type. Previously a `LineSeries` between two stacked `ColumnSeries` could mess up stacking for the latter.
- Zooming in on a part of `LineSeries` which had `strokeDashArray` enabled via `propertyFields` was making the whole line dashed.
- On a chart with `XYCursor` hovering on category with a null value for some series, used to display a tooltip for a different category in some cases.

[Full change log](https://github.com/amcharts/amcharts4/blob/master/dist/es2015/CHANGELOG.md).

Make sure you clear your browser cache after upgrading. And feel free to contact us again if you are still experiencing this issue.

**Comment by sassomedia** [comment]
Excellent, thanks! 🎉
